File: apps/android/app/src/main/java/com/karibuhealth/app/domain/LabCatalog.kt
package com.karibuhealth.app.domain

/** HC III lab catalog — exact names for deterministic ordering (migration 069). */
object LabCatalog {

    data class Test(val code: String, val name: String, val category: String)

    val tests = listOf(
        Test("MRDT", "Malaria RDT", "Malaria"),
        Test("BS_MPS", "Blood slide for malaria parasites", "Malaria"),
        Test("HIV_RDT", "HIV rapid test", "Serology"),
        Test("HB", "Haemoglobin", "Haematology"),
        Test("AFB", "Sputum smear (AFB / TB)", "Microbiology"),
        Test("URINALYSIS", "Urinalysis", "Urine"),
        Test("STOOL_OC", "Stool microscopy (ova/cysts)", "Microbiology"),
        Test("RBS", "Random blood sugar", "Biochemistry"),
        Test("SYPHILIS", "Syphilis test (RPR/TPHA)", "Serology"),
        Test("UCG", "Pregnancy test (UCG)", "Serology"),
        Test("WIDAL", "Widal test (typhoid)", "Serology"),
        Test("STOOL_RDT", "Stool antigen / H. pylori RDT", "Microbiology"),
    )

    fun byCategory(): List<Pair<String, List<Test>>> =
        tests.groupBy { it.category }.toList()

    fun nameForCode(code: String): String? = tests.find { it.code == code }?.name
}

File: packages/shared/src/lab-catalog.ts
/** HC III lab test catalog — mirrors lab_test_catalog (migration 069) so the
 *  web/Android order pickers send exact catalog names, never free text. */

export interface LabCatalogTest {
  code: string;
  name: string;
  category: string;
  specimen: string;
}

export const LAB_CATALOG_TESTS: LabCatalogTest[] = [
  { code: 'MRDT', name: 'Malaria RDT', category: 'Malaria', specimen: 'blood' },
  { code: 'BS_MPS', name: 'Blood slide for malaria parasites', category: 'Malaria', specimen: 'blood' },
  { code: 'HIV_RDT', name: 'HIV rapid test', category: 'Serology', specimen: 'blood' },
  { code: 'HBSAG', name: 'Hepatitis B rapid test (HBsAg)', category: 'Serology', specimen: 'blood' },
  { code: 'BRU_RDT', name: 'Brucellosis rapid test', category: 'Serology', specimen: 'blood' },
  { code: 'HB', name: 'Haemoglobin', category: 'Haematology', specimen: 'blood' },
  { code: 'AFB', name: 'Sputum smear (AFB / TB)', category: 'Microbiology', specimen: 'sputum' },
  { code: 'URINALYSIS', name: 'Urinalysis', category: 'Urine', specimen: 'urine' },
  { code: 'STOOL_OC', name: 'Stool microscopy (ova/cysts)', category: 'Microbiology', specimen: 'stool' },
  { code: 'RBS', name: 'Random blood sugar', category: 'Biochemistry', specimen: 'blood' },
  { code: 'SYPHILIS', name: 'Syphilis test (RPR/TPHA)', category: 'Serology', specimen: 'blood' },
  { code: 'UCG', name: 'Pregnancy test (UCG)', category: 'Serology', specimen: 'urine' },
  { code: 'WIDAL', name: 'Widal test (typhoid)', category: 'Serology', specimen: 'blood' },
  { code: 'STOOL_RDT', name: 'Stool antigen / H. pylori RDT', category: 'Microbiology', specimen: 'stool' },
];

export function labTestsByCategory(): [string, LabCatalogTest[]][] {
  const map = new Map<string, LabCatalogTest[]>();
  for (const t of LAB_CATALOG_TESTS) {
    const list = map.get(t.category) ?? [];
    list.push(t);
    map.set(t.category, list);
  }
  return Array.from(map.entries());
}
